common/src/test/java/com/itsaky/androidide/utils/PathTraversalTest.kt

```
@@ -0,0 +1,160 @@
+		val root = tempFolder.newFolder("real-project")
+		val outside = tempFolder.newFolder("outside")
+		File(outside, "secret.txt").writeText("secret")
+
+		val symlinkCreated =
+			try {
+				Files.createSymbolicLink(File(root, "evil").toPath(), outside.toPath())
+				true
+			} catch (e: UnsupportedOperationException) {
+				// The filesystem itself doesn't support symlinks (e.g. FAT32).
+				false
+			} catch (e: FileSystemException) {
```

detekt (1.23.8)

[warning] The caught exception is swallowed. The original exception could be lost.

(detekt.exceptions.SwallowedException)